### PR TITLE
Upgrade `dataprotection` API version from 2022-04-01 to 2023-05-01

### PR DESCRIPTION
**Pull request by sinbai**
Two points to note:

- Although the stable API 2023-08-01 for `dataprotection` is available , it is not yet public, therefore upgraded to 2023-05-01.

- Confirmed with the service team that [SnapshotStore](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fblob%2Fc8d98ad576b439cf4377d8d17122d2e9e831ff9e%2Fspecification%2Fdataprotection%2Fresource-manager%2FMicrosoft.DataProtection%2Fstable%2F2022-04-01%2Fdataprotection.json%23L6080C18-L6080C19&data=05%7C01%7Cv-elenaxin%40microsoft.com%7Ceb61f915897147aa0afe08dbef542b9a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638366917033338186%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=sL7s2uSoqACtHRK5SZks1Jld2qOVAd%2F2EpLTxM7vCk0%3D&reserved=0) was replaced with [OperationalStore](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fblob%2Fc8d98ad576b439cf4377d8d17122d2e9e831ff9e%2Fspecification%2Fdataprotection%2Fresource-manager%2FMicrosoft.DataProtection%2Fstable%2F2023-05-01%2Fdataprotection.json%23L7039C19-L7039C19&data=05%7C01%7Cv-elenaxin%40microsoft.com%7Ceb61f915897147aa0afe08dbef542b9a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638366917033346948%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=9Q4nw2g7Vc9PbNvHTe09897ooBJQAZYmLD3gNDhj9cQ%3D&reserved=0) . Since this is a breaking change, it should be updated from TF 4.0.

### CODE REVIEW

File: internal/services/dataprotection/data_protection_backup_vault_resource.go [hunk 1]
```diff
@@ -84,6 +74,32 @@ func resourceDataProtectionBackupVault() *pluginsdk.Resource {
 			"tags": tags.Schema(),
 		},
 	}
+
+	// Confirm with the service team that `SnapshotStore` has been replaced with `OperationalStore`.
```
Comment (jackofallops) [inline internal/services/dataprotection/data_protection_backup_vault_resource.go:78]: Is this a question that still needs answering by the service team?

File: internal/services/dataprotection/data_protection_backup_vault_resource.go [hunk 2]
```diff
@@ -84,6 +74,32 @@ func resourceDataProtectionBackupVault() *pluginsdk.Resource {
...
+	if !features.FourPointOhBeta() {
+		resource.Schema["datastore_type"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(backupvaults.StorageSettingStoreTypesArchiveStore),
+				"SnapshotStore",
+				string(backupvaults.StorageSettingStoreTypesVaultStore),
+			}, false),
```
Comment (jackofallops) [inline internal/services/dataprotection/data_protection_backup_vault_resource.go:88]: Is there a reason not to include the new `backupvaults.StorageSettingStoreTypesOperationalStore` here also? Or is there a reason to wait for 4.0 to support it?

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @sinbai - Thanks for this PR, just a couple of questions that need a look below.

Thanks

File: internal/services/dataprotection/data_protection_backup_vault_resource.go (hunk 2 above)
Comment (sinbai) [reply internal/services/dataprotection/data_protection_backup_vault_resource.go]: It's a good idea to include `backupvaults.StorageSettingStoreTypesOperationalStore`, the code has been updated. Could you please take another look?

**Comment by sinbai** [comment]
> Hi @sinbai - Thanks for this PR, just a couple of questions that need a look below.
> 
> Thanks

Hi @jackofallops thanks for your time. Could you please take another look?

**Review by katbyte — APPROVED** [review]
LGTM 🌵

**Pull request merged**